plugin/normalize.go

```diff
@@ -35,6 +35,21 @@ func (z Zones) Matches(qname string) string {
 	return zone
 }
 
+// dns.IsSubDomain is Inefficient methods
+func IsSubDomain(zone, qname string) bool {
```

Comment:
I think that community would benefit more if you would rather optimize the dns.IsSubDomain in https://github.com/miekg/dns than optimize it here